src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -641,12 +643,21 @@ void TemporalMemory::compute(const SDR &activeColumns, bool learn,
                              const SDR &extraWinners)
 {
   activateDendrites(learn, extraActive, extraWinners);
+
+  // Update Anomaly Metric.  The anomaly is the percent of active columns that
+  // were not predicted.
+  anomaly_ = nupic::algorithms::anomaly::computeRawAnomalyScore(
+                activeColumns,
+                cellsToColumns( getPredictiveCells() ));
+  // TODO: Update mean & standard deviation of anomaly here.
```

Comment:
please do not force a (specific) metric here. It's the same situation as with threshold,likelihood,... Let TM provide just the plain anomaly score in a correct, stable manner. And the users can do whatever they need with the an score.

Reply:
I'd like to provide both the plain anomaly score as well as two new `const&` attributes for `anomalyMean` and `anomalyStdDev`.  I want to provide these because I think that they are far more useful as a diagnostic tool than the raw anomaly score.  In a future PR I hope to overload `operator<<` to print the mean & stdDev of anomaly, among other diagnostic things.

Reply:
Can you do reasonably easily the experiment you want with anom mean/std computed manually? 
What I see as a problem is this complicates the primarily simple anomalyScore (parameter wise, also computational complexity). 

If you absolutely want the metrics be included, maybe AnomalyLikelihood is a better place (?)

Reply:
> What I see as a problem is this complicates the primarily simple anomalyScore (parameter wise, also computational complexity).

It should need 1 parameter, the time constant for the exponential moving averages.

It might not currently, but it should run in constant time.